doc/source/cluster/kubernetes/examples/distributed-checkpointing-with-gcsfuse.md

```diff
@@ -0,0 +1,264 @@
+(kuberay-distributed-checkpointing-gcsefuse)=
+
+# Distributed Checkpointing with KubeRay and GCSFuse
+
+This example demonstrates how to orchestrate distributed checkpointing with KubeRay, using the GCSFuse CSI driver and Google Cloud Storage as the remote storage system.
+To illustrate the concepts, this guide uses the [Finetuning a Pytorch Image Classifier with Ray Train](https://docs.ray.io/en/latest/train/examples/pytorch/pytorch_resnet_finetune.html) example.
+
+## Why Distributed Checkpointing with GCSFuse?
+
+In large-scale, high-performance machine learning, distributed checkpointing is crucial for fault tolerance, ensuring that if a node fails during training, the process can be resumed from the latest saved checkpoint instead of starting from scratch.
+While it's possible to directly reference remote storage paths (e.g., `gs://my-checkpoint-bucket`), using Google Cloud Storage FUSE (GCSFuse) offers distinct advantages for distributed applications. GCSFuse allows you to mount Cloud Storage buckets like local file systems, making checkpoint management more intuitive for distributed applications that rely on these semantics. Furthermore, GCSFuse is designed for high-performance workloads, delivering the performance and scalability needed for distributed checkpointing of large models.
+
+[Distributed checkpointing](https://docs.ray.io/en/latest/train/user-guides/checkpoints.html), in combination with [GCSFuse](https://cloud.google.com/storage/docs/gcs-fuse), allows for larger-scale model training with increased availability and efficiency.
+
+## Create a Kubernetes cluster on GKE
+
+Create a GKE cluster with the [GCSFuse CSI driver](https://cloud.google.com/kubernetes-engine/docs/how-to/persistent-volumes/cloud-storage-fuse-csi-driver) and [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) enabled, as well as a GPU node pool with 4 L4 GPUs:
+```
+export PROJECT_ID=<your project id>
```

Comment:
I think most users will know their PROJECT_ID because you need to create the GKE cluster. But PROJECT_NUMBER is kind of annoying to get. Unfortunately you need to know both. I added a reference to https://cloud.google.com/resource-manager/docs/creating-managing-projects#identifying_projects to make it easier to find it